File: backend/app/models/website.py
from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime, ForeignKey, Text, func
from sqlalchemy.orm import relationship
from app.database.database import Base


class Website(Base):
    __tablename__ = "websites"

    id = Column(Integer, primary_key=True, index=True)
    organization_id = Column(Integer, ForeignKey("organizations.id"), nullable=False)

    url = Column(Text, nullable=False)
    domain = Column(String(255), nullable=True)
    is_official = Column(Boolean, default=False)
    discovery_source = Column(String(255), nullable=True)
    confidence = Column(Float, default=0.0)
    status = Column(String(20), default="ACTIVE")  # ACTIVE/FAILED/BLOCKED/TIMEOUT
    error_message = Column(Text, nullable=True)
    created_at = Column(DateTime, server_default=func.now())

    organization = relationship("Organization", back_populates="websites")

File: backend/src/models/Website.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Website = sequelize.define('Website', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
    allowNull: false,
  },
  organization_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  url: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  domain: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  is_official: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
  discovery_source: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  confidence: {
    type: DataTypes.FLOAT,
    allowNull: true,
  },
  status: {
    type: DataTypes.STRING(20),
    defaultValue: 'ACTIVE',
  },
  error_message: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'websites',
  timestamps: false,
});

module.exports = Website;
